Objects/funcobject.c

```diff
@@ -395,12 +402,17 @@ _PyFunction_LookupByVersion(uint32_t version, PyObject **p_code)
         return slot->func;
     }
     return NULL;
+#else
+    return NULL;
 #endif
 }
 
 uint32_t
 _PyFunction_GetVersionForCurrentState(PyFunctionObject *func)
 {
+    // This function does not need locking/atomics as it can only be
```

Comment:
We use the function version in specialized instructions, so it is used in tier 1.
I'm surprised this isn't already synchronized.